File: pset6/prob/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # avg no of letters per 100 words
    avg_let = letters / words * 100
    # avg no of sentences per 100 words
    avg_words = sentences / words * 100

    # formula
    grade = round(0.0588 * avg_let - 0.296 * avg_words - 15.8)

    # print
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print("Grade " + str(grade))


def count_letters(text):
    l = 0
    for i in range(len(text)):
        if str.isalpha(text[i]):
            l += 1
    return l


def count_words(text):
    w = 1
    for i in range(len(text)):
        if str.isspace(text[i]):
            w += 1
    return w


def count_sentences(text):
    s = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            s += 1
    return s


if __name__ == "__main__":
    main()

File: pset2/prob/readability.c
//includes
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// declaration
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);



int main(void)
{
    // prompt input

    string text = get_string("Text: ");

    //count letters
    float x = count_letters(text);

    //count words
    float y = count_words(text);

    //count sentences
    float z = count_sentences(text);

    // find average
    // avg no of letters per 100 words
    float l = x / y * 100;

    // avg no of sentences per 100 words
    float s = z / y * 100;

    //using formula
    float index = 0.0588 * l - 0.296 * s - 15.8;
    int grade = round(index);


    //print result
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }


}



int count_letters(string text)
{
    int letters = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;

    for (int j = 0, n = strlen(text); j < n; j++)
    {
        if (isspace(text[j]))
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;

    for (int k = 0, n = strlen(text); k < n; k++)
    {
        if (text[k] == '.' || text[k] == '?' || text[k] == '!')
        {
            sentences++;
        }

    }
    return sentences;
}